internal/resources/aws/dynamodb_table.go

```diff
@@ -0,0 +1,366 @@
+package aws
+
+import (
+	"fmt"
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+)
+
+type DynamoDbTableArguments struct {
+	Address        string   `json:"address,omitempty"`
+	Region         string   `json:"region,omitempty"`
+	BillingMode    string   `json:"billingMode,omitempty"`
+	WriteCapacity  int64    `json:"writeCapacity,omitempty"`
+	ReadCapacity   int64    `json:"readCapacity,omitempty"`
+	ReplicaRegions []string `json:"replicaRegions,omitempty"`
+
+	MonthlyWriteRequestUnits       *int64 `json:"monthlyWriteRequestUnits,omitempty"`
+	MonthlyReadRequestUnits        *int64 `json:"monthlyReadRequestUnits,omitempty"`
+	StorageGB                      *int64 `json:"storageGB,omitempty"`
+	PitrBackupStorageGB            *int64 `json:"pitrBackupStorageGB,omitempty"`
+	OnDemandBackupStorageGB        *int64 `json:"onDemandBackupStorageGB,omitempty"`
+	MonthlyDataRestoredGB          *int64 `json:"monthlyDataRestoredGB,omitempty"`
+	MonthlyStreamsReadRequestUnits *int64 `json:"monthlyStreamsReadRequestUnits,omitempty"`
```

Comment:
Could we add the fields that @sinabakh mentioned in [his usage-file schema comment](https://github.com/infracost/infracost/issues/730#issuecomment-856993301) here so we have 1 source of truth for them? That source could then be used in the future by `--sync-usage-file` and docs.

Reply:
Yes, the usage schema for the resource is defined separately from the args a few lines below:
https://github.com/infracost/infracost/pull/803/files/ded0f66110486cb9aeffb5992fe7223aec699ba5#diff-71ed323ee96294a7b798471f490d0ecde86fa1b63763f1e023d4bf93e40dc21eR38-R46
and it's added to the resource created by NewDynamoDBTable.  (This is a little bit different than adding it to the RegistryItem which is what @sinabakh had proposed).

I made changes to sync-usage-file so it will use this resource.UsageSchema if present, or fall back to the "old way" of using the usage-example.